src/libraries/System.Collections.Immutable/src/System/Collections/Frozen/FrozenSet.cs

```diff
@@ -14,6 +15,40 @@ namespace System.Collections.Frozen
     /// </summary>
     public static class FrozenSet
     {
+        /// <summary>Creates a <see cref="FrozenSet{T}"/> with the specified values.</summary>
+        /// <param name="source">The values to use to populate the set.</param>
+        /// <typeparam name="T">The type of the values in the set.</typeparam>
+        /// <returns>A frozen set.</returns>
+        public static FrozenSet<T> Create<T>(params ReadOnlySpan<T> source) => Create(null, source);
+
+        /// <summary>Creates a <see cref="FrozenSet{T}"/> with the specified values.</summary>
+        /// <param name="source">The values to use to populate the set.</param>
+        /// <param name="equalityComparer">The comparer implementation to use to compare values for equality. If null, <see cref="EqualityComparer{T}.Default"/> is used.</param>
+        /// <typeparam name="T">The type of the values in the set.</typeparam>
+        /// <returns>A frozen set.</returns>
+        public static FrozenSet<T> Create<T>(IEqualityComparer<T>? equalityComparer, params ReadOnlySpan<T> source)
+        {
+            if (source.Length == 0)
+            {
+                return equalityComparer is null || ReferenceEquals(equalityComparer, FrozenSet<T>.Empty.Comparer) ?
+                    FrozenSet<T>.Empty :
+                    new EmptyFrozenSet<T>(equalityComparer);
+            }
+
+            HashSet<T> set =
```

Comment:
Not really related to this PR, but with all the improvements around `ROSpan<T>` (params, the new language support that makes `ROSpan<T>` the preferred overload, collection expression, etc) do we have any plans to expose constructor overloads on the various collection types so this can just be `new(source)`, much like we could if it had been an `IEnumerable<T>`?

Seems like a generally good thing that can simplify the patterns for common user scenarios. -- Notable doing `HashSet<T> set = [..source];` works as well, its just not super efficient today and looks to go through the enumerator not the special `for (...)` iteration that span normally gets.

Reply:
We'd like to, but we can't do that well yet. Even with how the current first-class span support is defined, if HashSet gained a `HashSet<T>(ReadOnlySpan<T>)` overload and you tried to call that with an array, it would fail to compile due to ambiguity, since array can implicitly convert to both `IEnumerable<T>` and `ReadOnlySpan<T>`.

This _should_ however resolve itself if/when we make `ReadOnlySpan<T> : IEnumerable<T>`, as that will result in `ReadOnlySpan<T>` being considered better for overload resolution. The ref struct work just landed in Roslyn main today. Once that lands in runtime and I can more easily vet it, my plan is to create a proposal for the resulting improvements we should make in .NET 9, including where we want to annotate with allow ref struct (e.g. delegates, enumerable) and what new APIs we can then expose that we've been wanting to (e.g. that ctor on HashSet).